iOS/Airbnb_iOS/Airbnb_iOS/Utility/DataSource/DetailPageCollectionDataSource.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  DetailPageCollectionDataSource.swift
+//  Airbnb_iOS
+//
+//  Created by juntaek.oh on 2022/06/07.
+//
+
+import UIKit
+
+class DetailPageCollectionDataSource: NSObject, UICollectionViewDataSource {
+
+    private var data = DetailPageItem()
+    private(set) var delegateObject: DetailPageViewController?
```

Comment:
다른 부분에서도 발생하지만 delegate 패턴 사용할 때도 객체 간의 참조 관계는 꼭 신경써주세요. 

Xcode Debug Memory Graph 이용하거나 간단한 방법으로 `deinit { print("Deinit") }` 이 호출되고 있지 않다면 메모리 릭이 발생하고 있습니다. 지금 코드 구조상 DetailPageCollectionDataSource 레퍼런스 카운트가 남아있을거 같아요.

```suggestion
    private(set) weak var delegateObject: DetailPageViewController?
```